swatch-contracts/src/main/java/com/redhat/swatch/contract/model/MeasurementMetricIdTransformer.java

```diff
@@ -144,14 +142,9 @@ private void checkForUnsupportedMetrics(List<Metric> metrics, SubscriptionEntity
       // Will check for correct metrics before translating awsDimension to Metrics UOM
       log.debug("Checking for unsupported metricIds");
       var supportedSet = metrics.stream().map(Metric::getAwsDimension).collect(Collectors.toSet());
-      List<SubscriptionMeasurementEntity> supportedMeasurements = new ArrayList<>();
-      // compare set of metric.awsDimension to measurements.metricId (pre Transformation state)
-      for (SubscriptionMeasurementEntity sub : subscription.getSubscriptionMeasurements()) {
-        if (supportedSet.contains(sub.getMetricId())) {
-          supportedMeasurements.add(sub);
-        }
-      }
-      subscription.setSubscriptionMeasurements(supportedMeasurements);
+      subscription
+          .getSubscriptionMeasurements()
+          .removeIf(m -> !supportedSet.contains(m.getMetricId()));
```

Comment:
I had to change this because I was getting this exception: 
```
hibernate - A collection with cascade=”all-delete-orphan” was no longer referenced by the owning entity instance
```
We should never create a new collection, but using and maintain the existing one which is referenced by the entity.